swap/chain/txscheduler.go

```diff
@@ -0,0 +1,127 @@
+package chain
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// TxSchedulerBackend is an extension of the normal Backend interface
+type TxSchedulerBackend interface {
+	Backend
+	// SendTransactionWithID is the same as SendTransaction but with the ID of the associated request passed alongside
+	// This is primarily used so the backend can react with the expected result during testing
+	SendTransactionWithID(ctx context.Context, id uint64, tx *types.Transaction) error
+}
+
+// DefaultTxSchedulerBackend is the standard backend that should be used
+// It simply wraps another Backend
+type DefaultTxSchedulerBackend struct {
+	Backend
+}
+
+// SendTransactionWithID in the default Backend calls the underlying SendTransaction function
+func (b *DefaultTxSchedulerBackend) SendTransactionWithID(ctx context.Context, id uint64, tx *types.Transaction) error {
+	return b.Backend.SendTransaction(ctx, tx)
+}
+
+// TxRequest describes a request for a transaction that can be scheduled
+type TxRequest struct {
+	To       common.Address // recipient of the transaction
+	Data     []byte         // transaction data
+	GasPrice *big.Int       // gas price or nil if suggested gas price should be used
+	GasLimit uint64         // gas limit or 0 if it should be estimated
+	Value    *big.Int       // amount of wei to send
+}
+
+// ToSignedTx returns a signed types.Transaction for the given request and nonce
+func (request *TxRequest) ToSignedTx(nonce uint64, opts *bind.TransactOpts) (*types.Transaction, error) {
+	tx := types.NewTransaction(
+		nonce,
+		request.To,
+		request.Value,
+		request.GasLimit,
+		request.GasPrice,
+		request.Data,
+	)
+
+	return opts.Signer(&types.HomesteadSigner{}, opts.From, tx)
```

Comment:
Yes, that's how this is supposed to be called. There was a change in signing from `Frontier` to `Homestead` but it has remained the same since, hence why the `HomesteadSigner` is still used today.

Reply:
Ok!